src/server.c

```diff
@@ -4124,6 +4125,14 @@ int processCommand(client *c) {
         }
     }
 
+    /* Check if the command has cross slot keys, incompatible with cluster mode. */
+    if (server.cluster_compatibility_sample_ratio && !server.cluster_enabled &&
```

Comment:
i realize that if we do want to maintain the counter also in cluster mode, this would become an `else` but could be complicated because we may still want to handle mustObeyClient?
or maybe we don't have to, or we don't care about this extra feature...
WDYT?

Reply:
yes, if we want to find incompatible operations when cluster is enabled, we may need to check in a slightly forward position, i.e. before the position where we may reply error in cluster mode. And i am not sure the meaning we do that, users want to know if they did  incompatible operations in cluster mode?

Reply:
Let's drop that idea. Not part of the requirements